### ISSUE TITLE
Wrapping and +/- operators

### ISSUE THREAD

**Issue by yole** [android/kotlin-guides#51]
The current guide says: "When a line is broken at a non-assignment operator the break comes before the symbol". Unfortunately this breaks semantics for +/- operators, which can be used in unary form. For example, this is a string concatenation:

    val s = "abc" +
        "def"

This is a simple string initialization, followed by an unused expression that applies unary + to a string literal:

    val s = "abc"
        + "def"

The wrapping rules should be changed to either break after all operators (preferably), or to make exceptions for +/-.

**Comment by eventualbuddha** [comment]
Anyone have thoughts on what should be done here?

**Comment by Jacksgong** [comment]
I search the checkstyle rule on SourceForge doc for [OperatorWrap](http://checkstyle.sourceforge.net/config_whitespace.html#OperatorWrap), as default, the [WrapOp]((http://checkstyle.sourceforge.net/property_types.html#wrapOp) ) for this rule is using NL, as it said on the [checkstyle repo](https://github.com/checkstyle/checkstyle), [google checkstyle rule](https://github.com/checkstyle/checkstyle/blob/80365b80228157a07cad060cfd30087487f4b6f9/src/main/resources/google_checks.xml) and [sun](https://github.com/checkstyle/checkstyle/blob/396a285549e89f618bab45b987c350dd1326379c/src/main/resources/sun_checks.xml) also use NL too, So our project also follow this rule for java code, so as the OperatorWrap default and google or sun defined for this, the correct option is:

```kotlin
val s = "abc"
     + "def"
if (candition1
     && candition2)
```

Although the above codestyle rule is for Java, but our project is develop with both Java and Kotlin, so if the rule is different, it is very confusing for our developers..

**Comment by yole** [comment]
@Jacksgong This is not a code style issue, this is a language syntax issue. The confusion of your developers is unfortunate, but you do have to wrap after `+` and `-`, not before them, if you want the code to function correctly.

**Comment by Jacksgong** [comment]
@yole okay, thanks for your response, I know that now, maybe because of there isn't `;` for kotlin lang, so for the case of 

```kotlin
val a = 1
       -2
```

we can't know whether `a` is `1` or `-1` now, it very confuses.

**Comment by yole** [comment]
I'm not sure what you mean by "we can't know". You can know this; `a` will be 1.

**Comment by Jacksgong** [comment]
Yep, I mean as the expert kotlin lang engineer, they absolutely know what does this means, but for the beginner, especially from Java, they will confusing, because on the Java, the following code is means `a` is `-1`:

```java
int a = 1
       -2;
```
